lua/nvim-lsp-installer/jobs/outdated-servers/git.lua

```diff
@@ -0,0 +1,42 @@
+local process = require "nvim-lsp-installer.process"
+local CheckResult = require "nvim-lsp-installer.jobs.outdated-servers.check-result"
+
+---@param server Server
+---@param source InstallReceiptSource
+---@param on_check_complete fun(result: CheckResult)
+return function(server, source, on_check_complete)
+    process.spawn("git", {
+        -- We assume git installation track the remote HEAD branch
+        args = { "fetch", "origin", "HEAD" },
```

Comment:
I'm not sure that this is reliable, since you might be starting with a shallow clone, so you might want to pass a `--force` flag. An alternative that I've found working relatively well for me is using [git ls-remote](https://git-scm.com/docs/git-ls-remote.html), which removes the hassle of dealing with `--unshallow` or deepening the history of a giant repo. 

Here's an example
https://github.com/LunarVim/LunarVim/blob/b10e18a213741dcd4ce1e7ba63accaf10719dbec/lua/lvim/utils/git.lua#L146-L154

Reply:
Yeah it does work, hopefully there are no gotchas related to the installed version of git.

Reply:
> Yeah it does work, hopefully there are no gotchas related to the installed version of git.

I have a solution in store if this is the case:
![608p65](https://user-images.githubusercontent.com/6705160/148391140-e64813c4-3c51-4b15-9950-81b2e21ab4cc.jpeg)